core/src/main/java/com/linecorp/armeria/server/annotation/JacksonRequestConverterFunction.java

```diff
@@ -74,6 +74,12 @@ public Object convertRequest(ServiceRequestContext ctx, AggregatedHttpMessage re
                 try {
                     return reader.readValue(content);
                 } catch (JsonProcessingException e) {
+                    if (expectedResultType == byte[].class ||
+                        expectedResultType == HttpData.class ||
+                        expectedResultType == String.class) {
```

Comment:
Oh, we missed `CharSequence.class`. 😅 It is also handled by `StringRequestConverterFunction`, so it should be checked from here, too.
And, would you please add the test about that type, too?